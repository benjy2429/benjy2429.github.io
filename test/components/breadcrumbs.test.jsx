/**
 * @jest-environment jsdom
 */

import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Breadcrumbs from '../../src/components/breadcrumbs';

describe('Breadcrumbs component', () => {
  it('does not render with no crumbs', () => {
    const component = shallow(<Breadcrumbs />);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('renders crumbs with links', () => {
    const crumbs = [{ name: 'Crumb', href: '/crumb' }];
    const component = shallow(<Breadcrumbs crumbs={crumbs} />);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('renders crumbs without links', () => {
    const crumbs = [{ name: 'Crumb' }];
    const component = shallow(<Breadcrumbs crumbs={crumbs} />);
    expect(toJson(component)).toMatchSnapshot();
  });
});
