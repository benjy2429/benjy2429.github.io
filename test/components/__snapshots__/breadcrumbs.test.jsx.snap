// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumbs component does not render with no crumbs 1`] = `""`;

exports[`Breadcrumbs component renders crumbs with links 1`] = `
<ul
  className="breadcrumbs"
>
  <li
    className="breadcrumb"
  >
    <GatsbyLink
      to="/"
    >
      Home
    </GatsbyLink>
  </li>
  <li
    className="breadcrumb"
    key="crumb-Crumb"
  >
    <GatsbyLink
      to="/crumb"
    >
      Crumb
    </GatsbyLink>
  </li>
</ul>
`;

exports[`Breadcrumbs component renders crumbs without links 1`] = `
<ul
  className="breadcrumbs"
>
  <li
    className="breadcrumb"
  >
    <GatsbyLink
      to="/"
    >
      Home
    </GatsbyLink>
  </li>
  <li
    className="breadcrumb"
    key="crumb-Crumb"
  >
    Crumb
  </li>
</ul>
`;
